frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -763,30 +763,9 @@ Tap a book in the search results to open it.]]),
     for id, common_setting in pairs(dofile("frontend/ui/elements/common_info_menu_table.lua")) do
         self.menu_items[id] = common_setting
     end
-    self.menu_items.exit_menu = {
-        text = _("Exit"),
-        hold_callback = function()
-            self:exitOrRestart()
-        end,
-    }
-    self.menu_items.exit = {
-        text = _("Exit"),
-        callback = function()
-            self:exitOrRestart()
-        end,
-    }
-    self.menu_items.restart_koreader = {
-        text = _("Restart KOReader"),
-        callback = function()
-            self:exitOrRestart(function()
-                UIManager:restartKOReader()
-            end)
-        end,
-    }
-    if not Device:canRestart() then
-        self.menu_items.exit_menu = self.menu_items.exit
-        self.menu_items.exit = nil
-        self.menu_items.restart_koreader = nil
+    -- insert common exit for filemanager
+    for id, common_setting in pairs(dofile("frontend/ui/elements/common_exit_menu.lua").table(self)) do
+        self.menu_items[id] = common_setting
```

Comment:
This feels like a new idiom, having to do this to pass self/app (but I have nothing against new stuff :)
Our current way of solving this is to not pass self/app, and use `UIManager:broadcastEvent(...)` so the event can be caught by the module that has `:onExitOrRestart()`. Dunno if that's possible here.

Reply:
> Our current way of solving this is to not pass self/app, and use UIManager:broadcastEvent(...) so the event can be caught by the module that has :onExitOrRestart(). Dunno if that's possible here.

Also because then it can easily tie in with gestures and such. I've had to rewrite a bunch of stuff for that, which was using an idiom more along these lines previously.